app/forms/firewall-rules-common.tsx

```diff
@@ -272,7 +278,7 @@ export const CommonFields = ({ control, nameTaken, error }: CommonFieldsProps) =
     if (!type || !value) return
     if (targets.value.some((t) => t.value === value && t.type === type)) return
     targets.onChange([...targets.value, { type, value }])
-    targetForm.reset()
+    targetForm.reset(targetAndHostDefaultValues)
```

Comment:
I wasn't seeing this reset do what I expected, and I figured out why: we call this reset inside the callback to `handleSubmit`. This does in fact set `isSubmitted` to false. But after it runs our callback, [`handleSubmit` sets `isSubmitted: true`](https://github.com/react-hook-form/react-hook-form/blob/9a497a70accbc5e0be5f4e1c06c0ff00baa963da/src/logic/createFormControl.ts#L1194-L1203). Hence the very brief flash of `false`, then back to `true`.

<img src="https://github.com/user-attachments/assets/ddda2fa3-2570-43c3-9f38-26f0d085469f" width=300>

Reply:
Fixed by `useEffect` below.